packages/library/src/rpc-websocket-autopinger.ts

```diff
@@ -45,9 +46,28 @@ export function getWebSocketTransportWithAutoping({ intervalMs, transport }: Con
                 } finally {
                     pingableConnections.delete(connection);
                     clearInterval(intervalId);
+                    if (handleOffline && handleOnline) {
+                        globalThis.window.removeEventListener('offline', handleOffline);
+                        globalThis.window.removeEventListener('online', handleOnline);
+                    }
```

Comment:
Is it worth splitting this into 2 ifs, one for each listener? The current code always defines them together, but it'd probably be easy to refactor it so that's not guaranteed and we might not notice we need to update this

Reply:
Good call. This is one of those ‘no really TypeScript trust me these will always be defined’ things, but none the less it's worth splitting them up for future protection against refactoring as you've mentioned!